test/functional/test-amp-context.js

```diff
@@ -61,6 +59,43 @@ describe('3p ampcontext.js', () => {
     windowMessageHandler = undefined;
   });
 
+  it('should send message when report3pError_()', () => {
+    win.name = generateSerializedAttributes();
+    const context = new AmpContext(win);
+    expect(context).to.be.ok;
+
+    // Resetting since a message is sent on construction.
+    windowPostMessageSpy.reset();
+
+    const e = new Error();
+    e.message = 'message';
+
+    const messagePayload = {
+      sentinel: '1-291921',
+      type: MessageType.USER_ERROR,
+    };
+
+    const messageData = 'amp-' + JSON.stringify(messagePayload);
+    const message = {
+      source: context.client_.hostWindow_,
+      data: messageData,
+      message: e.message,
+    };
+    windowMessageHandler(message);
+
+    window.onerror = function(message, source, lineno, colno, error) {
+      expect(error).to.equal(e);
+      expect(message).to.equal('message');
+      context.report3pError_();
+      expect(windowPostMessageSpy).to.be.called;
+      expect(windowPostMessageSpy).to.be.calledWith(
```

Comment:
please use `.jsonObject` as we agreed.

Reply:
I just took a look at the code, postMessage would call to serializeMessage which will JSON.stringify(message). Therefore I think checking the value as string here is expected.

Reply:
you can use `deserializeMessage` before testing. I know the test works fine, but it would be really hard for people to fix, if it fails for any reason.